docs(readme): update to reflect shared types

mdonoughe opened the pull request:
## Problem

_What problem are you trying to solve? What issue does this close?_

New developers may not be successful setting up their development environment.

Closes #220 

## Solution

_How did you solve the problem?_

This PR is just some documentation updates for creating a development environment.

**Improvements**:

- Specified that Docker is required for `npm run dev`
- Documented a potential issue that may be encountered when using Node.js 17

**Bug Fixes**:

- Fixed spelling of `audit-dep`
- Added missing `npm run build-frontend` in `npm run dev`. Otherwise, the server immediately exits with an error about the client index.html not existing.
- Added missing `npm run build` before `npm run server`. This is for the same reason as adding `npm run build-frontend`. It unnecessarily builds the backend, but it's simpler for somebody reading the readme.

## Tests

_What tests should be run to confirm functionality?_

Running through the instructions to set up a development environment should go more smoothly.

mdonoughe committed:
fix spelling of audit-dep

mdonoughe committed:
call out Docker dependency

mdonoughe committed:
build is required before server

mdonoughe committed:
document openssl error

mdonoughe committed:
build client as part of dev command

LoneRifle commented on package.json:
```diff
@@ -11,7 +11,7 @@
     "server": "npm run dev --prefix server",
     "client": "npm start --prefix client",
-    "dev": "concurrently \"./wait-for-it.sh localhost:3306 -t 0 -- npm run build-shared && sleep 5 && npm run server\" \"npm run client\" \"docker-compose up \"",
+    "dev": "concurrently \"./wait-for-it.sh localhost:3306 -t 0 -- npm run build-shared && npm run build-frontend && sleep 5 && npm run server\" \"npm run client\" \"docker-compose up \"",
```
```suggestion
    "dev": "concurrently \"./wait-for-it.sh localhost:3306 -t 0 -- npm run build-shared && sleep 5 && npm run server\" \"npm run client\" \"docker-compose up \"",
```
Building the frontend isn't actually needed, since that's covered by `npm run client`

mdonoughe replied:
I thought it would be too.

`npm run client` does build the client, but it doesn't write the files to `client/build`, and even though the frontend hosted by the `npm run client` server will be used, the backend will fail to start if `client/build/index.html` doesn't exist. If you delete that directory and then do `npm run client` in one terminal, wait for it to finish, and then do `npm run server` in another terminal, you will get an error about not being able to find the client files.

Alternatively, if `server/src/bootstrap/index.ts` somehow knew it was running in a development setup it could skip trying to read `index.html`.

LinHuiqing replied:
Thanks for bringing this up - I'll be doing the latter soon 👍

LoneRifle reviewed (changes requested):
lgtm otherwise. Thanks for the PR!

LoneRifle commented on README.md:
```diff
@@ -90,10 +90,10 @@ Optionally [DBeaver](https://dbeaver.io/download/) to view database with GUI
 
   # for backend server only
-  npm run server 
+  npm run build && npm run server
```
```suggestion
  npm run build-shared && npm run server
```

LoneRifle committed:
chore: incorporate suggestions from review

- revert 846a6e5, since `npm run client` hot-builds the frontend
- use `build-shared` before running server, not `build`
